### PR TITLE
Lint CI enforcement and cargo fmt

### PR DESCRIPTION
**Pull request by kjvalencik**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+        rust-toolchain: [nightly]
+
+    steps:
```
Comment (dherman) [inline .github/workflows/lint.yml:22]: I think it's worth adding a comment explaining why we have to do such an explosion of different clippy invocations here.

**Review by dherman — APPROVED** [review]
Just one suggestion for adding a comment.

Feels so good to get all these lints and standardizations enforced! I've wanted to standardize the codebase for years. Thank you!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (kjvalencik) [reply .github/workflows/lint.yml:22]: I found the cargo package features RFC which provides a way to selectively enable feature flags in a workspace on some packages.

It's working really well. I was able to add `cargo clippy-legacy` and `cargo clippy-core` aliases.

https://doc.rust-lang.org/cargo/reference/unstable.html#package-features

**Review by dherman — APPROVED** [review]
Great improvement! :shipit:

**Pull request merged**